azurerm/config.go

```diff
@@ -370,12 +371,16 @@ func setUserAgent(client *autorest.Client) {
 		client.UserAgent = fmt.Sprintf("%s %s", client.UserAgent, azureAgent)
 	}
 
+	if partnerID != "" {
+		client.UserAgent = fmt.Sprintf("%s pid-%s", client.UserAgent, partnerID)
+	}
+
 	log.Printf("[DEBUG] AzureRM Client User Agent: %s\n", client.UserAgent)
 }
 
 // getArmClient is a helper method which returns a fully instantiated
 // *ArmClient based on the Config's current settings.
-func getArmClient(c *authentication.Config, skipProviderRegistration bool) (*ArmClient, error) {
+func getArmClient(c *authentication.Config, skipProviderRegistration bool, partnerId string) (*ArmClient, error) {
```

Comment:
Since you always pass empty value to `partner_id` in test code, I feel you actually don't need to change getArmClient here.

Reply:
getArmClient is used within the live provider and the test code - so this is necessary (whilst we could probably update the additional wrapper, we build clients in a couple of different manners depending on the requirements)

Reply:
@tombuildsstuff you're correct, I missed the usage `provider.go` while reading it late night.